src/crewai/knowledge/source/base_file_knowledge_source.py

```diff
@@ -59,13 +70,31 @@ def convert_to_path(self, path: Union[Path, str]) -> Path:
 
     def _process_file_paths(self) -> List[Path]:
         """Convert file_path to a list of Path objects."""
-        paths = (
-            [self.file_path]
-            if isinstance(self.file_path, (str, Path))
-            else self.file_path
-        )
+
+        # Check if old file_path is being used
+        if hasattr(self, "file_path") and self.file_path is not None:
+            self._logger.log(
+                "warning",
+                "The 'file_path' attribute is deprecated and will be removed in a future version. Please use 'file_paths' instead.",
+                color="yellow",
+            )
+            paths = (
+                [self.file_path]
+                if isinstance(self.file_path, (str, Path))
+                else self.file_path
+            )
+        else:
+            if self.file_paths is None:
+                raise ValueError("Your source must be provided with a file_paths: []")
+            paths = (
+                [self.file_paths]
+                if isinstance(self.file_paths, (str, Path))
+                else self.file_paths
+            )
 
         if not isinstance(paths, list):
```

Comment:
We can drop with the new way you've set this up!

Reply:
fixed redundancy